compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -3500,18 +3500,14 @@ private void validateAnnotationAttachmentExpr(BLangAnnotationAttachment annAttac
             return;
         }
 
-
-        // At this point the type is a subtype of  map<anydata>|record{ anydata...; } or
-        // map<anydata>[]|record{ anydata...; }[], thus an expression is required.
         if (annAttachmentNode.expr == null) {
-            this.dlog.error(annAttachmentNode.pos, DiagnosticErrorCode.ANNOTATION_ATTACHMENT_REQUIRES_A_VALUE,
-                            annotationSymbol);
-            return;
+            annAttachmentNode.expr = new BLangRecordLiteral();
```

Comment:
I'm not sure if we should be adding this here (we usually defer modifying the AST until we get to desugar).

Also, wouldn't it be confusing to have this go in the normal type checking path for a mapping constructor, since there's no visible mapping constructor, here? How about checking if the type is a record with required fields (that do not have defaults) here and logging a specific error like

```
annotation value expected for annotation of record type ''{0}'' with required fields
```